cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -547,3 +556,30 @@ const (
 	// EncryptionAlgorithmRSA defines the RSA encryption algorithm type.
 	EncryptionAlgorithmRSA EncryptionAlgorithmType = "RSA"
 )
+
+// Timeouts holds various timeouts that apply to kubeadm commands.
+type Timeouts struct {
+	// ControlPlaneComponentHealthCheck is the amount of time to wait for a control plane
+	// component, such as the API server, to be healthy during "kubeadm init" and "kubeadm join".
```

Comment:
Should we mention the default value here?

Reply:
i have been waiting for API machinery to add the `+default=xx` feature, but looks like it's still WIP.

we are not consistent about showing defaults in godoc comments in the kubeadm's API. users have to use `config print xx-defaults` to see them. might be a good idea to put all the defaults as part of this PR.

Reply:
updated to mention defaults in the v1beta4 public type